cvat/apps/events/handlers.py

```diff
@@ -0,0 +1,378 @@
+# Copyright (C) 2023 CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from copy import deepcopy
+from datetime import datetime, timezone
+
+from rest_framework.renderers import JSONRenderer
+from crum import get_current_user, get_current_request
+
+from cvat.apps.engine.models import (
+    Organization,
+    Project,
+    Task,
+    Job,
+    User,
+    CloudStorage,
+    Issue,
+    Comment,
+    Label,
+)
+from cvat.apps.engine.serializers import (
+    ProjectReadSerializer,
+    TaskReadSerializer,
+    JobReadSerializer,
+    BasicUserSerializer,
+    CloudStorageReadSerializer,
+    IssueReadSerializer,
+    CommentReadSerializer,
+    LabelSerializer,
+)
+from cvat.apps.engine.models import ShapeType
+from cvat.apps.organizations.serializers import OrganizationReadSerializer
+from cvat.apps.webhooks.signals import project_id, organization_id
+from cvat.apps.engine.log import vlogger
+
+from .event import event_scope, create_event
+
+
+def task_id(instance):
+    if isinstance(instance, Task):
+        return instance.id
+
+    try:
+        tid = getattr(instance, "task_id", None)
+        if tid is None:
+            return instance.get_task_id()
+        return tid
+    except Exception:
+        return None
+
+def job_id(instance):
+    if isinstance(instance, Job):
+        return instance.id
+
+    return None
+
+def _get_current_user(instance):
+    if isinstance(instance, User):
+        return instance
+
+    if isinstance(instance, Job):
+        return instance.segment.task.owner
+
+    return get_current_user()
+
+def user_id(instance):
+    current_user = _get_current_user(instance)
```

Comment:
Why do we need to call _get_current_user? We handle a request. The request has an authentication user. Why should we get a user from job.segment.task.owner?